### ISSUE TITLE
Advanced search - Failing SQL query

### ISSUE THREAD

**Issue by Noodleyman** [cubecart/v6#2468]
catalogue.class.php, line 1960

`$q2 = "SELECT I.* FROM ".$GLOBALS['config']->get('config', 'dbprefix')."CubeCart_inventory AS I LEFT JOIN (SELECT product_id, MAX(price) as price, MAX(sale_price) as sale_price FROM ".$GLOBALS['config']->get('config', 'dbprefix')."CubeCart_pricing_group $group_id GROUP BY product_id) as G ON G.product_id = I.product_id $joinString WHERE I.product_id IN (SELECT product_id FROM `".$GLOBALS['config']->get('config', 'dbprefix')."CubeCart_category_index` as CI INNER JOIN ".$GLOBALS['config']->get('config', 'dbprefix')."CubeCart_category as C where CI.cat_id = C.cat_id AND C.status = 1) AND I.status = 1 ".$whereString.$like;`

evaluates SQL query to be:
```
SELECT
   I.* 
FROM
   cc_CubeCart_inventory AS I 
   LEFT JOIN
      (
         SELECT
            product_id,
            MAX(price) as price,
            MAX(sale_price) as sale_price 
         FROM
            cc_CubeCart_pricing_group 
         WHERE
            group_id = 0 
         GROUP BY
            product_id
      )
      as G 
      ON G.product_id = I.product_id 
WHERE
   I.product_id IN 
   (
      SELECT
         product_id 
      FROM
         `cc_CubeCart_category_index` as CI 
         INNER JOIN
            cc_CubeCart_category as C 
      where
         CI.cat_id = C.cat_id 
         AND C.status = 1
   )
   AND I.status = 1 IF (G.product_id IS NULL, IF (I.sale_price IS NULL 
   OR I.sale_price = 0, I.price, I.sale_price) >= 0.952, IF (G.sale_price IS NULL 
   OR G.sale_price = 0, G.price, G.sale_price) >= 0.952) IF (G.product_id IS NULL, IF (I.sale_price IS NULL 
   OR I.sale_price = 0, I.price, I.sale_price) <= 1.05, IF (G.sale_price IS NULL 
   OR G.sale_price = 0, G.price, G.sale_price) <= 1.05) 
   AND I.featured = '1' 
   AND `live_from` < UNIX_TIMESTAMP() 
   AND `live_from` < UNIX_TIMESTAMP() 
   AND 
   (
      I.name RLIKE '[[:<:]]1[[:>:]]' 
      OR I.description RLIKE '[[:<:]]1[[:>:]]' 
      OR I.product_code RLIKE '[[:<:]]1[[:>:]]'
   )
ORDER BY
   `name` ASC LIMIT 12 OFFSET 0
```

#1064 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IF (G.product_id IS NULL, IF (I.sale_price IS NULL
   OR I.sale_price = 0, I.p' at line 33

**Comment by abrookbanks** [comment]
What a mess. I never authored this code so it's a struggle.

**Comment by Noodleyman** [comment]
I've never liked the way the search was put together, recycling the category code. It could be massively simplified and segmented from the category product collection methods for backwards compatibility. :) 

Every time I am asked to review a change to search I cry a little...

**Comment by abrookbanks** [comment]
Christ its a total shambles.

**Comment by abrookbanks** [comment]
But I fixed this one..... :ok_hand:

**Comment by abrookbanks** [comment]
Pretty confident this fixed #2469 and #2470 too but its hard to tell what the advanced search parameters were.
